### DISCUSSION TITLE
Help us to improve MQTTX!

### DISCUSSION THREAD

**Discussion by oceanlvr** [emqx/MQTTX#452] [Ideas]
🚀 Feature request 🚀 

THANKS FOR YOUR SUPPORT!  🤗 🤗 

Description: Suggest an idea for improving `MQTT X` 

Labels: type: Enhancement

```markdown
### Is your proposal related to a problem?

<!--
  Provide a clear and concise description of what the problem is.
  For example, "I'm always frustrated when..."
-->

(Write your answer here.)

### Describe the solution or feature you'd like

<!--
  Provide a clear and concise description of what you want to happen.
-->

(Describe your proposed solution here.)


### Additional context

<!--
  Is there anything else you can add about the proposal?
  You might want to link to related issues here, if you haven't already.
-->

(Write your answer here.)
```

**Comment by oceanlvr** [comment]
Please write down the features you need, discuss them with us, and help us improve.

You can write down your thoughts below this discussion!

**Reply by tafo86awtech** [reply]
The possibility to modify the subscriptions would be useful, now I can only create a new one. 
For example in my case I have many topics that change for only word.
Also it is possible that I make a mistake and I don't want create a new subscription to change a character.

**Reply by zjn0505** [reply]
> The possibility to modify the subscriptions would be useful, now I can only create a new one. For example in my case I have many topics that change for only word. Also it is possible that I make a mistake and I don't want create a new subscription to change a character.

+1
It will also be convenient to group subscription, to enable and disable the subscriptions in 1 click.

**Reply by ysfscream** [reply]
In the latest version: https://github.com/emqx/MQTTX/releases/tag/v1.7.3

Editing of topics and `enable/disable` topic features are already available.

<img width="252" alt="image" src="https://user-images.githubusercontent.com/21299158/168259102-eb1a918c-f8ab-40ee-be0a-d000dcbfaba1.png">

**Reply by ysfscream** [reply]
> It will also be convenient to group subscription, to enable and disable the subscriptions in 1 click.

We think there is a lot of room for optimization for subscription lists.

**Comment by tafo86awtech** [comment]
Hi, Why is not possible save subscripton?
It would be a useful feature.

**Reply by ysfscream** [reply]
Hi, thanks for your suggestion, we are optimizing some operations on Topic and will implement quick configuration of subscriptions and quick selection of published Topics in the next few releases.

**Comment by 10D24D** [comment]
Hi。MQTTX真的太好用了，同时还想完善以下情况：
1. 除了MQTTX主界面，连接、订阅、消息这三个界面的宽度目前还不支持拖拉进行调整。建议增加界面调节功能。
2. MQTTX主界面的比例调整之后，消息界面的内容宽度不会自适应主界面的宽度。建议增加界面适配功能。
3. 连接界面，点击一个连接后会把其他已收缩的分组自动展开。这里不应该影响到其他分组的展开/收缩的显示状态。
4. 连接的配置界面，只有在连接的时候才可以保存连接信息显得比较鸡肋。建议增加一个保存配置的操作按钮。
5. 订阅界面，删除一个话题后，消息界面仍然存在该话题的历史数据。建议增加询问是否清空数据，或者在设置中进行配置。

**Reply by ysfscream** [reply]
Hi, we are very happy about your comments! Your suggestions are good and we will incorporate them into a follow-up plan to start optimizing these sections.

**Comment by PietjeNL** [comment]
### Is your proposal related to a problem?

For testing we use a couple of the same payloads to publish to different topics. (different devices)
Every time i need to rewrite or copy/paste the messages, or search the last few send.
Mostly it is for getting sensor data to check if all is installed and working ok.

### Describe the solution or feature you'd like

Create buttons with predefined payloads to be send to the topic.
Maybe a good place is under Connections, so it is available.
With drag and drop to reorder them.

### Additional context
When the button with the predefined payload is clicked put the payload in the field, don't touch the topic.

**Reply by ysfscream** [reply]
Hi, I think your suggestions would be helpful. I will create a new issue to track this issue, and maybe we will implement MQTT X Flow later (automated testing features can be satisfied, or the data for each connection and topic will be independent)

But sorry about that. However, we need to wait for a stable version of MQTT X to be completed before implementing it, and there are currently some performance issues.

**Reply by ChrisJamesonGroves** [reply]
I was going to suggest this exact same feature. We do the same, for testing, its often useful to send different payloads/topics and its a bit painful to keep updating the publish contents.

So how about having an additional column, much like the subscriptions column you currently have. Then with the current publish area there should be an additional button to 'Save Publish Definition' which adds this to the Publish Column. Then the user can just click on it to publish that same topics/payload.

**Comment by CptHolzschnauz** [comment]
I like to propose again to have a special treatment for $SYS/#.
With MQTTX it's still very hard to monitor the $SYS/broker. IMHO it would be clever to handle these data under "Connections", as a part of the broker.

To visualize my idea (Imagine the left (from MQTT Explorer) under the broker in "Connections" in MQTTX 

<img width="593" alt="Bildschirmfoto 2023-11-05 um 16 36 38" src="https://github.com/emqx/MQTTX/assets/68701579/cdafde4f-63f2-4697-845c-a73f0910c4fc">

That is the only missing feature keeping me with MQTT Explorer.

**Comment by ScottAtTrivedi** [comment]
I'd like to suggest some improvements for the UI, I'm currently using v1.10.1:

-drag subscription topics to rearrange

-When hovering over topics, the tag that appears gets in the way. Long hover should have tag appear, not immediately. 

-Topics should be able to be enabled or disabled without requiring to be connected first

-Clicking the vertical bar on the left side of the subscription topic element should toggle enable/disable

-Pressing enter on topic editing both adds newline and accepts changes. Newline character shouldn't be allowed in topics.
